src/python/gudhi/cubical_complex.pyx

```diff
@@ -20,20 +20,26 @@ import os
 import sys
 
 import numpy as np
+cimport numpy as np
 
 __author__ = "Vincent Rouvreau"
 __copyright__ = "Copyright (C) 2016 Inria"
 __license__ = "MIT"
 
+# Necessary because of PyArray_SimpleNewFromData
+np.import_array()
```

Comment:
I have the following warning with this modification:
```
In file included from /home/gailuron/miniconda3/envs/gudhi-dev/lib/python3.9/site-packages/numpy/core/include/numpy/ndarraytypes.h:1948,
                 from /home/gailuron/miniconda3/envs/gudhi-dev/lib/python3.9/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                 from /home/gailuron/miniconda3/envs/gudhi-dev/lib/python3.9/site-packages/numpy/core/include/numpy/arrayobject.h:5,
                 from /home/gailuron/workspace/gudhi/gudhi-devel/src/python/gudhi/cubical_complex.cpp:860:
/home/gailuron/miniconda3/envs/gudhi-dev/lib/python3.9/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
      |  ^~~~~~~
```

Seems to be a *normal* warning (cf. [this cython doc](https://docs.cython.org/en/latest/src/userguide/source_files_and_compilation.html#configuring-the-c-build)) that can be fixed with cython 3.0. Not to be fixed, but to keep in mind.